File: app/src/main/java/com/shentu/wallpager/model/HotPagerModel.java
package com.shentu.wallpager.mvp.model;

import android.app.Application;

import com.google.gson.Gson;
import com.jess.arms.di.scope.FragmentScope;
import com.jess.arms.integration.IRepositoryManager;
import com.shentu.wallpager.app.BasePageModel;
import com.shentu.wallpager.mvp.contract.HotPagerContract;
import com.shentu.wallpager.mvp.model.api.service.MicroService;
import com.shentu.wallpager.mvp.model.entity.Subject;
import com.shentu.wallpager.mvp.model.entity.SubjectsEntity;
import com.shentu.wallpager.mvp.model.entity.Wallpaper;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Observable;


@FragmentScope
public class HotPagerModel extends BasePageModel implements HotPagerContract.Model {

    @Inject
    Gson mGson;
    @Inject
    Application mApplication;

    @Inject
    public HotPagerModel(IRepositoryManager repositoryManager) {
        super(repositoryManager);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        this.mGson = null;
        this.mApplication = null;
    }

    @Override
    public Observable<SubjectsEntity> getSubjectList(boolean clear) {
        offset = clear ? 0 : (limit + offset);
        return mRepositoryManager
                .obtainRetrofitService(MicroService.class)
                .getSubjectList(limit, offset);
    }
}

File: app/src/main/java/com/shentu/wallpaper/mvp/model/HotPagerModel.kt
package com.shentu.wallpaper.mvp.model

import android.app.Application
import com.blankj.utilcode.util.NetworkUtils
import com.google.gson.Gson
import com.jess.arms.di.scope.FragmentScope
import com.jess.arms.integration.IRepositoryManager
import com.shentu.wallpaper.app.BasePageModel
import com.shentu.wallpaper.app.Constant
import com.shentu.wallpaper.model.api.cache.MicroCache
import com.shentu.wallpaper.model.api.service.MicroService
import com.shentu.wallpaper.model.response.BannerPageResponse
import com.shentu.wallpaper.model.response.WallpaperPageResponse
import com.shentu.wallpaper.mvp.contract.TabHomeContract
import io.reactivex.Observable
import io.rx_cache2.DynamicKey
import io.rx_cache2.EvictProvider
import javax.inject.Inject


@FragmentScope
class HotPagerModel @Inject
constructor(repositoryManager: IRepositoryManager) : BasePageModel(repositoryManager), TabHomeContract.Model {

    @Inject
    lateinit var mGson: Gson

    @Inject
    lateinit var mApplication: Application

    override fun getRecommends(clear: Boolean): Observable<WallpaperPageResponse> {
        offset = if (clear) 0 else limit + offset
        return mRepositoryManager
                .obtainRetrofitService(MicroService::class.java)
                .getRecommendWallpapers(limit, offset)
    }

    override fun getBanners(): Observable<BannerPageResponse> {
        return mRepositoryManager
                .obtainRetrofitService(MicroService::class.java)
                .getBanners()

    }
}